### DISCUSSION TITLE
Use with Existing Public Key

### DISCUSSION THREAD

**Discussion by requincreative** [spatie/crypto#36] [General]
I am attempting to perform a one-off encryption of client file and the client's vendor is insisting on using their existing Public Key available at the following URL: https://secure.nmi.com/merchants/Import-PGP.asc

When I attempt to create the object I get an Invalid Public Key exception

```
 $found = Storage::disk('public')->exists('pgp.asc');
if($found){

    $pgp_key = Storage::disk('public')->get('pgp.asc');
    $pgp = Spatie\Crypto\Rsa\PublicKey::fromString($pgp_key);
    dd($pgp);
}
```

Spatie \ Crypto \ Rsa \ Exceptions \ InvalidPublicKey
PHP 8.1.5
10.16.1
**This does not seem to be a valid public key.**

I've tried on Windows 10 / PHP 8.1.5 and Ubuntu/ PHP 8.1.5

**Comment by FullMe7alJacke7** [comment]
@requincreative Did you ever get this resolved? 

I have a similar requirement and I was looking into if I could use this package or not.
Thanks!
